drivers/flash/flash_stm32_ospi.c

```diff
@@ -0,0 +1,1498 @@
+	s_command.InstructionMode    = HAL_OSPI_INSTRUCTION_8_LINES;
+	s_command.InstructionDtrMode = HAL_OSPI_INSTRUCTION_DTR_DISABLE;
+	s_command.Instruction = SPI_NOR_OCMD_RESET_EN;
+	s_command.InstructionSize = HAL_OSPI_INSTRUCTION_16_BITS;
+	if (HAL_OSPI_Command(&dev_data->hospi,
+		&s_command, HAL_OSPI_TIMEOUT_DEFAULT_VALUE) != HAL_OK) {
+		LOG_ERR("OSPI reset enable (OPI/STR) failed");
+		return -EIO;
+	}
+
+	/* reset memory in OPI mode and STR transfer mode */
+	s_command.Instruction = SPI_NOR_OCMD_RESET_MEM;
+	if (HAL_OSPI_Command(&dev_data->hospi,
+		&s_command, HAL_OSPI_TIMEOUT_DEFAULT_VALUE) != HAL_OK) {
+		LOG_ERR("OSPI reset memory (OPI/STR) failed");
+		return -EIO;
+	}
+
+	/* reset enable in OPI mode and DTR transfer mode */
+	s_command.InstructionDtrMode = HAL_OSPI_INSTRUCTION_DTR_ENABLE;
+	s_command.Instruction = SPI_NOR_OCMD_RESET_EN;
+	if (HAL_OSPI_Command(&dev_data->hospi,
+		&s_command, HAL_OSPI_TIMEOUT_DEFAULT_VALUE) != HAL_OK) {
+		LOG_ERR("OSPI reset enable (OPI/DTR) failed");
+		return -EIO;
+	}
+
+	/* reset memory in OPI mode and DTR transfer mode */
+	s_command.Instruction = SPI_NOR_OCMD_RESET_MEM;
+	if (HAL_OSPI_Command(&dev_data->hospi,
+		&s_command, HAL_OSPI_TIMEOUT_DEFAULT_VALUE) != HAL_OK) {
+		LOG_ERR("OSPI reset memory (OPI/DTR) failed");
+		return -EIO;
+	}
+
+#endif
+	/* After SWreset CMD, wait in case SWReset occurred during erase operation */
+	k_msleep(STM32_OSPI_RESET_MAX_TIME);
+
+	return 0;
+}
+
+/*
+ * Function to erase the flash : chip or sector with possible OSPI/SPI and STR/DTR
+ * to erase the complete chip (using dedicated command) :
+ *   set size >= flash size
+ *   set addr = 0
+ */
+static int flash_stm32_ospi_erase(const struct device *dev, off_t addr,
+				  size_t size)
+{
+	const struct flash_stm32_ospi_config *dev_cfg = DEV_CFG(dev);
+	struct flash_stm32_ospi_data *dev_data = DEV_DATA(dev);
+	int ret = 0;
+
+	/* erase non-zero size */
+	if (size == 0) {
+		return 0;
+	}
+
+	/* Maximise erase size : means the complete chip */
+	if (size > dev_cfg->flash_size) {
+		/* reset the addr in that case */
+		addr = 0;
+		size = dev_cfg->flash_size;
+	}
+
+	if (!ospi_address_is_valid(dev, addr, size)) {
+		LOG_DBG("Error: address or size exceeds expected values: "
+			"addr 0x%lx, size %zu", (long)addr, size);
+		return -EINVAL;
+	}
+
+	if ((size != SPI_NOR_SECTOR_SIZE) && (size < dev_cfg->flash_size)) {
+		LOG_DBG("Error: wrong sector size 0x%x", size);
+		return -ENOTSUP;
+	}
+
+	OSPI_RegularCmdTypeDef cmd_erase = {
+		.OperationType = HAL_OSPI_OPTYPE_COMMON_CFG,
+		.FlashId = HAL_OSPI_FLASH_ID_1,
+		.AlternateBytesMode = HAL_OSPI_ALTERNATE_BYTES_NONE,
+		.DataMode = HAL_OSPI_DATA_NONE,
+		.DummyCycles = 0,
+		.DQSMode = HAL_OSPI_DQS_DISABLE,
+		.SIOOMode = HAL_OSPI_SIOO_INST_EVERY_CMD,
+	};
+
+	ospi_lock_thread(dev);
+
+	/* Check Flash busy */
+	if (stm32_ospi_mem_ready(&dev_data->hospi,
+		dev_cfg->data_mode, dev_cfg->data_rate) != 0) {
+		LOG_DBG("Erase failed : flash busy");
```

Comment:
```suggestion
		LOG_ERR("Erase failed : flash busy");
```
That looks important to be seen.